mlflow/models/container/__init__.py

```diff
@@ -76,13 +76,34 @@ def _serve():
         raise Exception("This container only supports models with the MLeap or PyFunc flavors.")
 
 
-def _install_pyfunc_deps(model_path=None, install_mlflow=False):
+def _install_pyfunc_deps(model_path=None, install_mlflow=False, no_conda=False):
     """
     Creates a conda env for serving the model at the specified path and installs almost all serving
     dependencies into the environment - MLflow is not installed as it's not available via conda.
     """
     # If model is a pyfunc model, create its conda env (even if it also has mleap flavor)
+    def install_dependency_from_condafile(env_path_dst):
```

Comment:
Could we achieve this by simply executing `["conda env update -f {env_path_dist}".format(env_path_dist=env_path_dist)]` rather than attempting to parse the conda env into a requirements.txt & installing that?

Reply:
Hi @smurching , Thank you for the feedback. Please find my thoughts below.

Usually conda.yaml file has a name field and in that case when we run "conda env update" command, it usually creates a new conda environment if it doesn't exists. Again it adds up more space and the docker image size would increase. Hence I feel `conda env update` will not work in this case.

However I can try `conda env update -n base` but not sure how it would work and i afraid if name field in conda.yaml would override the env name. I will do some experiments on this and update.

Reply:
Hi @ambiSanthu, yep exactly, the idea is to run `conda env update` while the `base` environment is activated - I'd tried this on the anaconda docker image while the base image was activated and found it to work, let me know if you hit any issues!

Reply:
If conda is a must, Can we use miniconda instead of anaconda?

Reply:
> Hi @ambiSanthu, yep exactly, the idea is to run `conda env update` while the `base` environment is activated - I'd tried this on the [anaconda docker image](https://hub.docker.com/r/continuumio/anaconda/) while the base image was activated and found it to work, let me know if you hit any issues!

But if conda.yaml file have "name" field then it creates a new conda environment with the name mentioned in conda.yaml file. Again it might not be an optimized image. In such cases, we might need to first remove the line that contains "name" in conda.yaml which I feel an ugly fix. Please share your thoughts on this @smurching

Reply:
> If conda is a must, Can we use miniconda instead of anaconda?

We are already using miniconda in docker templates.

Reply:
@smurching please let me know your thoughts on this.

> > Hi @ambiSanthu, yep exactly, the idea is to run `conda env update` while the `base` environment is activated - I'd tried this on the [anaconda docker image](https://hub.docker.com/r/continuumio/anaconda/) while the base image was activated and found it to work, let me know if you hit any issues!
> 
> But if conda.yaml file have "name" field then it creates a new conda environment with the name mentioned in conda.yaml file (even though "-n base" is mentioned). Again it might not be an optimized image. In such cases, we might need to first remove the line that contains "name" in conda.yaml which I feel an ugly fix. Please share your thoughts on this @smurching